src/query/interpreter.cpp

```diff
@@ -1252,7 +1282,10 @@ PreparedQuery Interpreter::PrepareTransactionQuery(std::string_view query_upper,
 
       in_explicit_transaction_ = true;
       expect_rollback_ = false;
-      metadata_ = GenOptional(metadata);
+      metadata_ = GenOptional(extras.metadata_pv);
+
+      auto const timeout = DetermineTxTimeout(extras.tx_timeout, interpreter_context_->config);
+      explicit_transaction_timer_ = timeout ? std::make_shared<utils::AsyncTimer>(timeout.value()) : nullptr;
```

Comment:
so this works because of
`explicit operator bool() const { return value_.has_value(); }` ?

Reply:
Yes. Even though it is explicit conversion, implicit bool conversion is allowed in `if`'s and ternary conditionals.